### PR TITLE
chore: polish code of ingestion server

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Resolve sonarqube scan issue of ingestion server

## Implementation highlights

Resolve scan issue of ingestion server, and did below test cases:
1. Deploy MSK sink
2. Deploy S3 sink
3. Update appIds and endpoint path

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [x] deploy ingestion server
    - [x] with MSK sink
    - [ ] with KDS sink
    - [x] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts [hunk 1]
```diff
@@ -40,6 +40,17 @@ interface ResourcePropertiesType {
   protocol: string;
 }
 
+interface HandleClickStreamSDKInput {
+  appIds: string;
```
Comment (zxkane) [inline src/ingestion-server/custom-resource/update-alb-rules/index.ts:44]: add `readonly` modifier if possible

File: src/ingestion-server/server/parameter.ts [hunk 2]
```diff
@@ -241,7 +241,7 @@ export function createStackParameters(scope: Construct, props: {deliverToKinesis
           'logS3Bucket and logS3Prefix cannot be empty when enableApplicationLoadBalancerAccessLog=Yes',
       },
     ],
-  });
+  }).overrideLogicalId('logS3BucketAndEnableLogRule');
```
Comment (zxkane) [inline src/ingestion-server/server/parameter.ts:244]: It's unnecessary, we already suppressed this false positive finding.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/ingestion-server/custom-resource/update-alb-rules/index.ts (hunk 1 above)
Comment (dengmingtong) [reply src/ingestion-server/custom-resource/update-alb-rules/index.ts]: Fixed

File: src/ingestion-server/server/parameter.ts (hunk 2 above)
Comment (dengmingtong) [reply src/ingestion-server/server/parameter.ts]: Rollbacked

**Review by zxkane — APPROVED** [review]

**Pull request merged**